api/v1beta1/spiaccesscheck_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SPIAccessCheckSpec defines the desired state of SPIAccessCheck
+type SPIAccessCheckSpec struct {
+	RepoUrl     string      `json:"repoUrl"`
+	Permissions Permissions `json:"permissions,omitempty"`
+}
+
+// SPIAccessCheckStatus defines the observed state of SPIAccessCheck
+type SPIAccessCheckStatus struct {
+	RepoURL         string                    `json:"repoURL"`
+	Accessible      bool                      `json:"accessible"`
+	Private         bool                      `json:"private,omitempty"`
```

Comment:
What about calling this `Accessibility` with an enum value of `public`, `private`, `non-existent` or `unknown`? `Accessible` still needs to remain there because it tells us whether the repo is accessible in this namespace using whatever tokens already present here.

Reply:
True. I'd personally still prefer `Accessibility: public|private|unknown` over `Private: true|false` but you're right it is more a matter of taste and documentation...

Reply:
done